### DISCUSSION TITLE
Adding custom events to Root element

### DISCUSSION THREAD

**Discussion by moy2010** [facebook/lexical#3613] [Ideas]
Before I work on a PR, I would like to hear your thoughts on the idea of adding custom events to the `Root` element; i.e. if this has been previously discussed internally (didn't find any relatable discussion on GitHub), etc.

The motivation behind this is that currently there's no way to extend the functionality for keyboard events, which cause some combinations to be consumed eagerly (i.e. `CTRL + B` is consumed by the "bold text keyDown" event, thus making it impossible to listen i.e. to the `CTRL + SHIFT + B` combination.

**Comment by trueadm** [comment]
You can listen to the FORMAT_TEXT_COMMAND at a higher priority than the built in listeners and intercept the event and check for those modifiers. You can return true in your listener to stop the command propagating more too :)

**Reply by moy2010** [reply]
Ah, good idea! I still think the idea of accepting custom event listeners is worth the shot, since currently the logic for handling such events would need to be spread in different places (i.e. some event listeners need to be placed on a parent node, and others need to be added to the update listener in the editor itself. Unless of course a special event is created so that all logic is handled in the update listener :thinking: .

**Reply by trueadm** [reply]
You can just add an event listener to the root element itself if needed, especially for those not handled by Lexical, in an effect if you're using React.

**Reply by moy2010** [reply]
Isn't `ContentEditable` the root element? If so, it doesn't accept an `onKeyDown` prop.
